Fixes shadowing warnings

sickpig opened the pull request:
Since #1171 and #1172 got merged, if the compiler supports it, a lot of warnings started to be raised during the build process.

This PR is taking care at the most obvious warnings generated by `-Wshadow`flag, e.g. : 

```
     connmgr.cpp:38:46: warning: declaration shadows a variable in the global namespace [-Wshadow]
    static bool RemoveNode(std::vector<CNode *> &vNodes, CNode *pNode)
                                                 ^
    ./net.h:191:29: note: previous declaration is here
    extern std::vector<CNode *> vNodes;
                                ^
```

sickpig committed:
Fix variables shadowing in stat.h

./stat.h:146:51: warning: declaration shadows a field of 'CStat<DataType, RecordType>' [-Wshadow]
    virtual UniValue GetSeries(const std::string &name, int count)
                                                  ^
./stat.h:83:17: note: previous declaration is here
    std::string name;

./stat.h:301:49: warning: declaration shadows a field of 'CStatHistory<DataType, RecordType>' [-Wshadow]
    int Series(int series, DataType *array, int len)
                                                ^
./stat.h:190:9: note: previous declaration is here
    int len[STATISTICS_NUM_RANGES];
        ^

sickpig committed:
Fix variables shadowing in torcontrol.cpp

torcontrol.cpp:206:25: warning: declaration shadows a field of 'TorControlConnection' [-Wshadow]
    const ConnectionCB &connected,
                        ^
torcontrol.cpp:105:51: note: previous declaration is here
    boost::function<void(TorControlConnection &)> connected;
                                                  ^
torcontrol.cpp:207:25: warning: declaration shadows a field of 'TorControlConnection' [-Wshadow]
    const ConnectionCB &disconnected)
                        ^
torcontrol.cpp:107:51: note: previous declaration is here
    boost::function<void(TorControlConnection &)> disconnected;
                                                  ^
torcontrol.cpp:455:56: warning: declaration shadows a field of 'TorController' [-Wshadow]
void TorController::add_onion_cb(TorControlConnection &conn, const TorControlReply &reply)
                                                       ^
torcontrol.cpp:391:26: note: previous declaration is here
    TorControlConnection conn;
                         ^

sickpig committed:
Fix variable shadowing in test/wallet_tests.cpp

wallet/test/wallet_tests.cpp:225:18: warning: declaration shadows a local variable [-Wshadow]
        for (int i = 0; i < 20; i++)
                 ^
wallet/test/wallet_tests.cpp:304:22: warning: declaration shadows a local variable [-Wshadow]
            for (int i = 0; i < RANDOM_REPEATS; i++)
                     ^
wallet/test/wallet_tests.cpp:321:22: warning: declaration shadows a local variable [-Wshadow]
            for (int i = 0; i < RANDOM_REPEATS; i++)
                     ^
wallet/test/wallet_tests.cpp:81:14: note: previous declaration is here
    for (int i = 0; i < RUN_TESTS; i++)
             ^

sickpig committed:
Fix variable shadowing in test/util_tests.cpp

test/util_tests.cpp:110:58: warning: declaration shadows a variable in namespace 'util_tests' [-Wshadow]
void ThreadCorralTest(CThreadCorral *c, int region, int *readVal, int setVal)
                                                         ^
test/util_tests.cpp:49:21: note: previous declaration is here
static volatile int readVal = 0;

sickpig committed:
Fix variable shadowing in rpcwallet.cpp

wallet/rpcwallet.cpp:1240:24: warning: declaration shadows a local variable [-Wshadow]
            tallyitem &item = mapAccountTally[strAccount];
                       ^
wallet/rpcwallet.cpp:1264:37: warning: declaration shadows a local variable [-Wshadow]
                for (const uint256 &item : (*it).second.txids)
                                    ^
wallet/rpcwallet.cpp:1220:61: note: previous declaration is here
    for (const std::pair<CTxDestination, CAddressBookData> &item : pwalletMain->mapAddressBook)

wallet/rpcwallet.cpp:2685:28: warning: declaration shadows a local variable [-Wshadow]
            CTxDestination address;
                           ^
wallet/rpcwallet.cpp:2675:24: note: previous declaration is here
        CTxDestination address;
                       ^
wallet/rpcwallet.cpp:2834:43: warning: declaration shadows a variable in the global namespace [-Wshadow]
void RegisterWalletRPCCommands(CRPCTable &tableRPC)
                                          ^
./rpc/server.h:176:18: note: previous declaration is here
extern CRPCTable tableRPC;

sickpig committed:
Fix variable shadowing in rcpdump.cpp

wallet/rpcdump.cpp:117:10: warning: declaration shadows a variable in the global namespace [-Wshadow]
    bool fRescan = true;
         ^
wallet/rpcdump.cpp:187:10: warning: declaration shadows a variable in the global namespace [-Wshadow]
    bool fRescan = true;
         ^
wallet/rpcdump.cpp:311:10: warning: declaration shadows a variable in the global namespace [-Wshadow]
    bool fRescan = true;
         ^
wallet/rpcdump.cpp:382:10: warning: declaration shadows a variable in the global namespace [-Wshadow]
    bool fRescan = true;
         ^

wallet/rpcdump.cpp:568:10: warning: declaration shadows a variable in the global namespace [-Wshadow]
    bool fRescan = true;
         ^
wallet/wallet.h:45:26: note: previous declaration is here
extern std::atomic<bool> fRescan;

sickpig committed:
Fix variables shadowing in wallet.cpp

wallet/wallet.cpp:1153:55: warning: declaration shadows a local variable [-Wshadow]
                    map<uint256, int>::const_iterator mi = pwallet->mapRequestCount.find(hashBlock);
                                                      ^
wallet/wallet.cpp:1145:47: note: previous declaration is here
            map<uint256, int>::const_iterator mi = pwallet->mapRequestCount.find(GetHash());
                                              ^
wallet/wallet.cpp:2929:30: warning: declaration shadows a local variable [-Wshadow]
    for (set<CTxDestination> grouping : groupings)
                             ^
wallet/wallet.cpp:2870:25: note: previous declaration is here
    set<CTxDestination> grouping;
                        ^

sickpig committed:
Fix variables shadowing in script_tests.cpp

test/script_tests.cpp:295:37: warning: declaration shadows a field of 'script_tests::(anonymous namespace)::TestBuilder' [-Wshadow]
    TestBuilder &Add(const CScript &script)
                                    ^
test/script_tests.cpp:246:13: note: previous declaration is here
    CScript script;
            ^
test/script_tests.cpp:1550:17: warning: declaration shadows a local variable [-Wshadow]
        CScript condScript = CScript() << vaddr << OP_DATASIGVERIFY;
                ^
test/script_tests.cpp:1562:17: warning: declaration shadows a local variable [-Wshadow]
        CScript condScript = CScript() << vaddr << OP_DATASIGVERIFY;
                ^
test/script_tests.cpp:1572:17: warning: declaration shadows a local variable [-Wshadow]
        CScript condScript = CScript() << OP_DATASIGVERIFY;
                ^
test/script_tests.cpp:1485:13: note: previous declaration is here
    CScript condScript = CScript() << ToByteVector(dataSigner.addr) << OP_DATASIGVERIFY;

sickpig committed:
Fix variable shadowing in test/test_bitcoin_fuzzy.cpp

test/test_bitcoin_fuzzy.cpp:460:34: warning: declaration shadows a local variable [-Wshadow]
                for (const auto &ft : registry_seq)
                                 ^
test/test_bitcoin_fuzzy.cpp:439:15: note: previous declaration is here
    FuzzTest *ft = nullptr;
              ^

sickpig committed:
Fix variable shadowing test/net_tests.cpp

test/net_tests.cpp:57:56: warning: declaration shadows a variable in the global namespace [-Wshadow]
CDataStream AddrmanToStream(CAddrManSerializationMock &addrman)
                                                       ^
./net.h:185:17: note: previous declaration is here
extern CAddrMan addrman;
                ^

sickpig committed:
Fix variables shadowing in test/crypto_tests.cpp

test/crypto_tests.cpp:146:13: warning: declaration shadows a local variable [-Wshadow]
        int size = enc.Encrypt(&sub[0], sub.size(), &subout[0]);
            ^
test/crypto_tests.cpp:191:13: warning: declaration shadows a local variable [-Wshadow]
        int size = enc.Encrypt(&sub[0], sub.size(), &subout[0]);
            ^
test/crypto_tests.cpp:173:9: note: previous declaration is here
    int size = enc.Encrypt(&in[0], in.size(), &realout[0]);
        ^

sickpig committed:
Fix variable shadowing wallet/walletdb.cpp

wallet/walletdb.cpp:930:48: warning: declaration shadows a local variable [-Wshadow]
                    map<string, int>::iterator mi = bitdb.mapFileUseCount.find(strFile);
                                               ^
wallet/walletdb.cpp:920:44: note: previous declaration is here
                map<string, int>::iterator mi = bitdb.mapFileUseCount.begin();
                                           ^

sickpig committed:
Fix variable shadowing in wallet/db.cpp

wallet/db.cpp:370:33: warning: declaration shadows a local variable [-Wshadow]
                            int ret = db.ReadAtCursor(pcursor, ssKey, ssValue, DB_NEXT);
                                ^
wallet/db.cpp:352:25: note: previous declaration is here
                    int ret = pdbCopy->open(nullptr, // Txn pointer
                        ^

sickpig committed:
Fix variables shadowing in requestManager.cpp

unlimited.cpp:1656:46: warning: declaration shadows a variable in the global namespace [-Wshadow]
void RegisterUnlimitedRPCCommands(CRPCTable &tableRPC)
                                             ^
./rpc/server.h:176:18: note: previous declaration is here
extern CRPCTable tableRPC;
                 ^
  CXX      libbitcoin_common_a-dstencode.o
requestManager.cpp:535:18: warning: declaration shadows a variable in the global namespace [-Wshadow]
    unsigned int blkReqRetryInterval = MIN_BLK_REQUEST_RETRY_INTERVAL;
                 ^
requestManager.cpp:54:14: note: previous declaration is here
unsigned int blkReqRetryInterval = MIN_BLK_REQUEST_RETRY_INTERVAL;
             ^
requestManager.cpp:536:18: warning: declaration shadows a variable in the global namespace [-Wshadow]
    unsigned int txReqRetryInterval = MIN_TX_REQUEST_RETRY_INTERVAL;
                 ^
requestManager.cpp:51:14: note: previous declaration is here
unsigned int txReqRetryInterval = MIN_TX_REQUEST_RETRY_INTERVAL;
             ^
requestManager.cpp:1173:25: warning: declaration shadows a local variable [-Wshadow]
            for (CNode *pnode : vNodes)
                        ^
requestManager.cpp:1050:71: note: previous declaration is here
bool CRequestManager::MarkBlockAsReceived(const uint256 &hash, CNode *pnode)

sickpig committed:
Fix variables shadowing in txmempool.cpp

txmempool.cpp:1200:17: warning: declaration shadows a local variable [-Wshadow]
    for (txiter it : stage)
                ^
txmempool.cpp:1190:64: note: previous declaration is here
    indexed_transaction_set::index<entry_time>::type::iterator it = mapTx.get<entry_time>().begin();
                                                               ^
txmempool.cpp:1334:35: warning: declaration shadows a local variable [-Wshadow]
            BOOST_FOREACH (txiter it, stage)
                                  ^
txmempool.cpp:1315:74: note: previous declaration is here
        indexed_transaction_set::index<descendant_score>::type::iterator it = mapTx.get<descendant_score>().begin();

sickpig committed:
Fix variable shadowing in thinblock.cpp

thinblock.cpp:408:14: warning: declaration shadows a local variable [-Wshadow]
        CInv inv(CInv(MSG_BLOCK, thinBlockTx.blockhash));
             ^
thinblock.cpp:299:10: note: previous declaration is here
    CInv inv(MSG_XTHINBLOCK, thinBlockTx.blockhash);

thinblock.cpp:694:10: warning: declaration shadows a field of 'CXThinBlock' [-Wshadow]
    bool collision = false;
         ^
./thinblock.h:61:10: note: previous declaration is here
    bool collision;
         ^

sickpig committed:
Fix variable shadowing in rpc/mining.cpp

rpc/mining.cpp:661:14: warning: declaration shadows a local variable [-Wshadow]
    for (int i = 0; i < (int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++i)
             ^
rpc/mining.cpp:614:9: note: previous declaration is here
    int i = 0;
        ^
rpc/mining.cpp:976:43: warning: declaration shadows a variable in the global namespace [-Wshadow]
void RegisterMiningRPCCommands(CRPCTable &tableRPC)
                                          ^
./rpc/server.h:176:18: note: previous declaration is here
extern CRPCTable tableRPC;
                 ^

sickpig committed:
Fix variables shadowing in main.cpp

main.cpp:2183:27: warning: declaration shadows a local variable [-Wshadow]
            CDiskBlockPos pos;
                          ^
main.cpp:1963:16: note: previous declaration is here
    CDiskTxPos pos(pindex->GetBlockPos(), GetSizeOfCompactSize(block.vtx.size()));
               ^

main.cpp:3883:18: warning: declaration shadows a variable in the global namespace [-Wshadow]
        uint64_t maxTxSize = 0;
                 ^
./unlimited.h:272:29: note: previous declaration is here
extern CTweak<unsigned int> maxTxSize;
                            ^
main.cpp:3998:71: warning: declaration shadows a local variable [-Wshadow]
                std::multimap<CBlockIndex *, CBlockIndex *>::iterator it = range.first;
                                                                      ^
main.cpp:3977:29: note: previous declaration is here
    for (BlockMap::iterator it = mapBlockIndex.begin(); it != mapBlockIndex.end(); ++it)
                            ^

sickpig committed:
Fix variable shadowing in miner.cpp

CTransactionRef BlockAssembler::coinbaseTx(const CScript &scriptPubKeyIn, int nHeight, CAmount nValue)
                                                                              ^
./miner.h:55:9: note: previous declaration is here
    int nHeight;
        ^

sickpig committed:
Fix variables shadowing in connmgr.cpp

connmgr.cpp:38:46: warning: declaration shadows a variable in the global namespace [-Wshadow]
static bool RemoveNode(std::vector<CNode *> &vNodes, CNode *pNode)
                                             ^
./net.h:191:29: note: previous declaration is here
extern std::vector<CNode *> vNodes;
                            ^

sickpig committed:
Fix variable shadowing in util.cpp

util.cpp:683:23: warning: declaration shadows a variable in the global namespace [-Wshadow]
void FileCommit(FILE *fileout)
                      ^
util.cpp:270:14: note: previous declaration is here
static FILE *fileout = NULL;
             ^

sickpig committed:
Fix variable shadowing in netbase.cpp

netbase.cpp:720:15: warning: declaration shadows a variable in the global namespace [-Wshadow]
    proxyType nameProxy;
              ^
netbase.cpp:41:18: note: previous declaration is here
extern proxyType nameProxy;
                 ^

sickpig commented:
Next step is a PR related to the warning raised by thread safety analysis.

ptschip commented on src/stat.h:
```diff
@@ -69,9 +69,9 @@ class CStatBase
     // Returns the historical or series data along with timestamp
-    virtual UniValue GetSeriesTime(const std::string &name, int count) = 0;
+    virtual UniValue GetSeriesTime(const std::string &namep, int count) = 0;
 };
```
why use &namep rather than &_name?   .. same question in a few other files.  If you mean it to be a pointer then the convention throughout bitcoin is to prefix with a 'p', so it would be pname (as we do for example with pnode), but in this case it's a reference so i would think &_name would be appropriate.

sickpig replied:
With the ending p I meant parameter, but you are right that it is ambiguous. Going to use a _ prefix in all these cases

awemany commented:
Nit: A couple commit message misspellings: 'varaible', 'variablesa', 'shadowinf'.

Then I wonder whether there's better namings available than adding a `2` for some of the fixes, but then I also lack inspiration for something adequate there.

Thank you for going through this!

sickpig commented:
Add WIP rag, need to fix some commit message and use a better criteria in naming variables.

sickpig commented:
taking out of WIP:
- rebased
- follow proper naming conventions as suggested by @ptschip 
- fixed typos in commit messages

sickpig committed:
Fix variable shadowing in pubkey.cpp

pubkey.cpp:324:55: warning: declaration shadows a field of 'CExtPubKey' [-Wshadow]
bool CExtPubKey::Derive(CExtPubKey &out, unsigned int nChild) const
                                                      ^
./pubkey.h:169:18: note: previous declaration is here
    unsigned int nChild;
                 ^

sickpig committed:
Fix variable shadowing in bitnodes.cpp

bitnodes.cpp:308:33: warning: declaration shadows a local variable [-Wshadow]
                const UniValue &result = find_value(o, "node");
                                ^
bitnodes.cpp:296:25: note: previous declaration is here
        const UniValue &result = find_value(reply, "results");

sickpig committed:
Fix variable shadowing in key.cpp

bitnodes.cpp:308:33: warning: declaration shadows a local variable [-Wshadow]
                const UniValue &result = find_value(o, "node");
                                ^
bitnodes.cpp:296:25: note: previous declaration is here
        const UniValue &result = find_value(reply, "results");

sickpig committed:
Fix variable shadowing in netaddress.cpp

netaddress.cpp:173:18: warning: declaration shadows a local variable [-Wshadow]
        uint32_t ipNone = INADDR_NONE;
                 ^
netaddress.cpp:162:19: note: previous declaration is here
    unsigned char ipNone[16] = {};
                  ^

sickpig commented:
Add a bunch more. There's another one but I'm not sure it is an actual issue, I think the analyzer got confused by `BOOST_SCOPE_EXIT` macro expansion. This what I'm referring to: 

```
 main.cpp:2001:27: warning: declaration shadows a variable in the global namespace [-Wshadow]                               
        BOOST_SCOPE_EXIT(&PV, &fParallel) { PV->SetLocks(fParallel); }                                                     
                          ^                                                                                                            
./parallel.h:186:45: note: previous declaration is here                                                                              
extern std::unique_ptr<CParallelValidation> PV; // Singleton class                                                              
                                            ^ 
```

paging @ptschip

sickpig committed:
Fix variable shadowing in main.cpp

main.cpp:2001:27: warning: declaration shadows a variable in the global namespace [-Wshadow]
        BOOST_SCOPE_EXIT(&PV, &fParallel) { PV->SetLocks(fParallel); }
                          ^
./parallel.h:186:45: note: previous declaration is here
extern std::unique_ptr<CParallelValidation> PV; // Singleton class
                                            ^

ptschip commented on src/util.cpp:
```diff
@@ -680,19 +680,19 @@ bool TryCreateDirectories(const fs::path &p)
 
-void FileCommit(FILE *fileout)
+void FileCommit(FILE *_fileout)
 {
```
I think this one could just be changed to pfileout

sickpig replied:
fair, going to change it

ptschip commented on src/main.cpp:
```diff
@@ -2181,8 +2181,8 @@ bool ConnectBlock(const CBlock &block,
-            CDiskBlockPos pos;
-            if (!FindUndoPos(state, pindex->nFile, pos, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
+            CDiskBlockPos pos2;
+            if (!FindUndoPos(state, pindex->nFile, pos2, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
```
possibly an nit, but using pos2 seems to say there there's another one around nearby but really the it's in FindUndoPos() and FindBlockPos()  ... that's where the shadowing is happening.  It think it would be clearer to use _pos in those functions. 

-bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos, unsigned int nAddSize)
+bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &_pos, unsigned int nAddSize)

sickpig replied:
According to `clang` analyzer the shadowin happen here:

sickpig replied:
According to `clang` analyzer the shadowing happen here: 

```
    main.cpp:2183:27: warning: declaration shadows a local variable [-Wshadow]
                CDiskBlockPos pos;
                              ^
    main.cpp:1963:16: note: previous declaration is here
        CDiskTxPos pos(pindex->GetBlockPos(), GetSizeOfCompactSize(block.vtx.size()));
                   ^
```

sickpig replied:
Going to change to `_pos` anyway, looks better that way.

ptschip commented on src/thinblock.cpp:
```diff
@@ -691,7 +691,7 @@ bool CXThinBlock::process(CNode *pfrom,
     int unnecessaryCount = 0;
-    bool collision = false;
+    bool collision2 = false;
     std::map<uint64_t, uint256> mapPartialTxHash;
```
not sure what this is shadowing...collision is local only and is not getting passed in?

sickpig replied:
```
    thinblock.cpp:694:10: warning: declaration shadows a field of 'CXThinBlock' [-Wshadow]
        bool collision = false;
             ^
    ./thinblock.h:61:10: note: previous declaration is here
        bool collision;
             ^
```

awemany commented:
@sickpig: On `BOOST_SCOPE_EXIT`, this might be relevant (and suggests you are right): https://svn.boost.org/trac10/ticket/11640

ptschip commented:
oh, duh, didn't see that... can we make it _collision  rather than with a 2?

On 13/07/2018 5:27 AM, Andrea Suisani wrote:
>
> *@sickpig* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In src/thinblock.cpp
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1176#discussion_r202333260>:
>
> > @@ -691,7 +691,7 @@ bool CXThinBlock::process(CNode *pfrom,
>      // for a collision.
>      int missingCount = 0;
>      int unnecessaryCount = 0;
> -    bool collision = false;
> +    bool collision2 = false;
>      std::map<uint64_t, uint256> mapPartialTxHash;
> |thinblock.cpp:694:10: warning: declaration shadows a field of
> 'CXThinBlock' [-Wshadow] bool collision = false; ^
> ./thinblock.h:61:10: note: previous declaration is here bool
> collision; ^ |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1176#discussion_r202333260>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMRF0GsoKw-rpoYu24Jzg-XYz6hQbyjcks5uGJI3gaJpZM4U80HK>.
>

sickpig commented:
@ptschip sure no problem, will change it

ptschip commented on src/test/util_tests.cpp:
```diff
@@ -107,10 +107,10 @@ BOOST_AUTO_TEST_CASE(util_sharedcriticalsection)
 
 
-void ThreadCorralTest(CThreadCorral *c, int region, int *readVal, int setVal)
+void ThreadCorralTest(CThreadCorral *c, int region, int *_readVal, int setVal)
```
maybe use pReadVal

sickpig replied:
fair, will change.

sickpig commented:
@awemany WRT BOOST_SCOPE_EXIT shadow warning I applied a fix suggested by @ptschip, see: 9efac68735

That said according to this [comment](https://svn.boost.org/trac10/timeline?from=2018-06-19T11%3A38%3A58Z&precision=second) in the thread you mention above it seems that warning not a false positive, thou.

ptschip commented:
yes you're right

On 13/07/2018 5:36 AM, Andrea Suisani wrote:
>
> *@sickpig* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In src/main.cpp
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1176#discussion_r202335291>:
>
> > @@ -2181,8 +2181,8 @@ bool ConnectBlock(const CBlock &block,
>      {
>          if (pindex->GetUndoPos().IsNull())
>          {
> -            CDiskBlockPos pos;
> -            if (!FindUndoPos(state, pindex->nFile, pos, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
> +            CDiskBlockPos pos2;
> +            if (!FindUndoPos(state, pindex->nFile, pos2, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
>
> According to |clang| analyzer the shadowing happen here:
>
> |main.cpp:2183:27: warning: declaration shadows a local variable
> [-Wshadow] CDiskBlockPos pos; ^ main.cpp:1963:16: note: previous
> declaration is here CDiskTxPos pos(pindex->GetBlockPos(),
> GetSizeOfCompactSize(block.vtx.size())); ^ |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1176#discussion_r202335291>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMRF0Jy16NGJ-IfmeMrvcP4Upb146p9rks5uGJRTgaJpZM4U80HK>.
>

sickpig commented:
Fix all issue raised in the last run of reviews.

ptschip reviewed (approved):
Looks good.
